content/en/BestPractices/prevent-security-misconfig.md

```diff
@@ -0,0 +1,160 @@
+
+To prevent this:
+
+- Don't expose a stack trace to users
+- Configure custom error messages for your software
+     - For a .NET application, modify the `Web.config` file:
+
+          ```
+          <system.web>
+          <customErrors mode="RemoteOnly" defaultRedirect="Error.aspx">
+          <error statusCode="401" redirect="CustomPage.aspx" />
+          <error statusCode="403" redirect="CustomPage.aspx" />
+          <error statusCode="404" redirect="NotFound.aspx" />
+          <error statusCode="500" redirect="Error.aspx" />
+          </customErrors>
+          </system.web>
+          ```
+
+### Directory Listing
+
+Directory listing allows users to see all files and directories stored on a web server. A malicious actor may get unauthorized access to sensitive files. Search the web with `intitle: "index of /"` to see websites that have directory listing enabled.
+
+In the following example, we show how to prevent this problem for an Apache Tomcat web server:
+
+1. Open the `conf/web.xml` file in the directory where Tomcat is installed.
+
+     - In Windows 10, the default installation directory is `C:\Program Files (x86)\Apache Software Foundation\Tomcat 9.0`.
+
+1. In the file, set `<param-value>false</param-value>` to `false` for `<param-name>listings</param-name>`:
+
+     ```
+     <servlet>
+          <servlet-name>default</servlet-name>
+          <servlet-class>org.apache.catalina.servlets.DefaultServlet</servlet-class>
+          <init-param>
+               <param-name>debug</param-name>
+               <param-value>0</param-value>
+          </init-param>
+          <init-param>
+               <param-name>listings</param-name>
+               <param-value>false</param-value>
+          </init-param>
+          <load-on-startup>1</load-on-startup>
+     </servlet>
+     ```
+
+1. In the Apache web server, create a `.htaccess` file in the related directory.
+
+1. Add the following lines to the `httpd.conf` file, or replace existing lines with:
+
+     ```
+     <Directory /{YOUR DIRECTORY}>
+     Options FollowSymLinks
+     </Directory>
+     ```
+
+To turn directory listing off, modify the configuration files of your web server.
+
+### Default Accounts
+
+A developer may forget to remove default accounts when releasing the application.
+
+Examples of default accounts:
+
+- Application accounts
+- Web server logins
+- Cloud instances
+- Databases
+
+To prevent this:
+
+- Delete default accounts from the development environment once the development is complete
+- Change default passwords that come with installation packages
+- Restrict the exposure of administrator pages and web server logins in the production environment
+- Establish a strong password policy for your accounts
+
+### Files or Directories in a URL Path
+
+Sensitive files that may appear in a URL path include:
+
+- Backup files
+- Log files
+- Database files
```

Comment:
@shadixit, could an attacker also try to access config files with the same "brute force" methods? If so, I might add one line:

```suggestion
- Database files
- Configuration files
```

Reply:
Yes if configuration files with sensitive data are left in webroot, e.g. .env files, it can be picked up by URL bruteforcing. It is fine to add this.